### PR TITLE
21 string functions for comparing first and last characters of a string

### PR DESCRIPTION
**Pull request by adjiap**
### SUMMARY
Implements necessary changes for #21 

*Reviewers:* @numere-org/maintainers

### IMPLEMENTATION
* Implementation: Add beginswith() and endswith() string function with aliases, startswith(STR,STR), STR.startsw() and STR.endsw() respectively-
* Implementation test: 
```
X = "Hello World"
startswith(X, "Hel")  # True
endswith(X, "rld") # True
endswith(X, "asdf") # False
X.startsw("Hel")  # True
X.endsw("rld")    # True
```

### DOCUMENTATION
* [x] ChangesLog updated
* [x] Code changes commented
* **Documentation articles:**
    * [x] corresponding documentation articles updated
    * [ ] new documentation articles created
    * [ ] not needed
* **Language files:**
    * [x] corresponding language files updated (Updated in separate PR, https://github.com/numere-org/Translation/pull/17
    * [ ] not needed

----------------

### TESTS BY REVIEWERS
* [x] Added to the automatic SW tests
* [ ] Tested manually

### CODE REVIEW

File: changes.tmp [hunk 1]
```diff
@@ -16,3 +16,4 @@ Added	Folding has been enabled for LaTeX files.
 Added	The terminal now also proposed global variables for auto completion.
 Applied	The revision of large files won't be created by calculating a DIFF. Instead, the whole file content is saved.
 Added   It is possible to directly access the first and last character of a string
+Added	String functions to compare the first and last characters of a string to a substring
```
Comment (numeredev) [inline changes.tmp:19]:
```suggestion
Added	String functions to compare the first and last characters of a string to a substring: "startswith()" and "endswith()"
```

File: kernel/core/strings/stringfunctions.hpp [hunk 2]
```diff
@@ -2760,6 +2760,40 @@ static StringVector strfnc_sha256(StringFuncArgs& funcArgs)
...
+/// \return StringVector
+///
+/////////////////////////////////////////////////
+static StringVector strfnc_beginswith(StringFuncArgs& funcArgs)
+{
+    return funcArgs.sArg2.view().length() <= funcArgs.sArg1.view().length()
+        && equal(funcArgs.sArg2.view().begin(),
+                 funcArgs.sArg2.view().end(),
+                 funcArgs.sArg1.view().begin()
+                 );
```
Comment (numeredev) [inline kernel/core/strings/stringfunctions.hpp:2776]: Interesting method. A bit hard to read, I'm a bit more familiar with `funcArgs.sArg2.view() == funcArgs.sArg1.view().subview(0, funcArgs.sArg2.view().length())`, but that seems to work as well.

File: kernel/core/strings/stringfunctions.hpp [hunk 3]
```diff
@@ -2791,9 +2825,11 @@ static std::map<std::string, StringFuncHandle> getStringFuncHandles()
     mHandleTable["and"]                 = StringFuncHandle(VAL, strfnc_and, true);
     mHandleTable["ascii"]               = StringFuncHandle(STR, strfnc_ascii, false);
     mHandleTable["basetodec"]           = StringFuncHandle(STR_STR, strfnc_basetodec, false);
+    mHandleTable["beginswith"]          = StringFuncHandle(STR_STR, strfnc_beginswith, false);
```
Comment (numeredev) [inline kernel/core/strings/stringfunctions.hpp:2828]: Could probably be removed, can't it?

File: kernel/core/utils/tools.cpp [hunk 4]
```diff
@@ -3966,6 +3966,10 @@ static void replaceAccessMethods(string& sLine, size_t nPos, size_t nFinalPos, c
         sLine.replace(nPos, nFinalPos-nPos, "firstch(" + sReplacement + ")");
     else if (sMethod == "last")
         sLine.replace(nPos, nFinalPos-nPos, "lastch(" + sReplacement + ")");
+    else if (sMethod == "startsw")
+        sLine.replace(nPos, nFinalPos-nPos, "beginswith" + sArgument );
```
Comment (numeredev) [inline kernel/core/utils/tools.cpp:3970]:
```suggestion
        sLine.replace(nPos, nFinalPos-nPos, "startswith" + sArgument );

```

**Review by numeredev — CHANGES_REQUESTED** [review]
Only some small changes required here. Language strings seem to be missing.

File: kernel/core/strings/stringfunctions.hpp (hunk 3 above)
Comment (adjiap) [reply kernel/core/strings/stringfunctions.hpp:2828]: I realized that in the original requirement, #21 asks for beginswith() as the main, and startswith() as a possible alias(), so I left the beginswith() (well, added it back in)

**Comment by adjiap** [comment]
@numeredev Should there be an Issue as well in [Translation](https://github.com/numere-org/Translation) that I can make a feature branch out of? Or should I commit the changes some other name as a feature branch?

File: kernel/core/strings/stringfunctions.hpp (hunk 3 above)
Comment (numeredev) [reply kernel/core/strings/stringfunctions.hpp:2828]: `startswith()` is enough. `beginswith()` does not add any reasonable benefit here.

**Comment by numeredev** [comment]
> @numeredev Should there be an Issue as well in [Translation](https://github.com/numere-org/Translation) that I can make a feature branch out of? Or should I commit the changes some other name as a feature branch?

Easiest way is the following: create a branch in numere-org/Translation with the same name as this one, i.e. `21-string-functions-for-comparing-first-and-last-characters-of-a-string`. Then open up a pull request and the issue ID #21 in this pull request (second line or so).

File: kernel/core/strings/stringfunctions.hpp (hunk 3 above)
Comment (adjiap) [reply kernel/core/strings/stringfunctions.hpp]: Done

**Review by numeredev — CHANGES_REQUESTED** [review]
Code quality is fine, however there's a "glitch" in the code as follows:

This line fails in the tests:
`assert !startswith(sString, "") && !endswith(sString, "");`

You want to test, wether the comparison string actually has a length

**Review by numeredev — APPROVED** [review]
Tests ran successfully, changes accepted

**Pull request merged**